chainerrl/agents/a2c.py

```diff
@@ -0,0 +1,283 @@
+        self.v_loss_coef = v_loss_coef
+        self.entropy_coeff = entropy_coeff
+
+        self.average_actor_loss_decay = average_actor_loss_decay
+        self.average_value_decay = average_value_decay
+        self.average_entropy_decay = average_entropy_decay
+        self.batch_states = batch_states
+
+        self.xp = self.model.xp
+        self.t = 0
+        self.t_start = 0
+
+        # Stats
+        self.average_actor_loss = 0
+        self.average_value = 0
+        self.average_entropy = 0
+
+    def _flush_storage(self, obs_shape, action):
+        obs_shape = obs_shape[1:]
+        action_shape = action.shape[1:]
+
+        self.states = self.xp.zeros(
+            [self.update_steps + 1, self.num_processes] + list(obs_shape),
+            dtype='f')
+        self.actions = self.xp.zeros(
+            [self.update_steps, self.num_processes] + list(action_shape),
+            dtype=action.dtype)
+        self.rewards = self.xp.zeros(
+            (self.update_steps, self.num_processes, 1), dtype='f')
+        self.value_preds = self.xp.zeros(
+            (self.update_steps + 1, self.num_processes, 1), dtype='f')
+        self.returns = self.xp.zeros(
+            (self.update_steps + 1, self.num_processes, 1), dtype='f')
+        self.masks = self.xp.ones(
+            (self.update_steps, self.num_processes, 1), dtype='f')
+
+        self.obs_shape = obs_shape
+        self.action_shape = action_shape
+
+    def _compute_returns(self, next_value):
+        if self.use_gae:
+            self.value_preds[-1] = next_value
+            gae = 0
+            for i in reversed(range(self.update_steps)):
+                delta = self.rewards[i] + \
+                    self.gamma * self.value_preds[i + 1] * self.masks[i] - \
+                    self.value_preds[i]
+                gae = delta + self.gamma * self.tau * self.masks[i] * gae
+                self.returns[i] = gae + self.value_preds[i]
+        else:
+            self.returns[-1] = next_value
+            for i in reversed(range(self.update_steps)):
+                self.returns[i] = self.rewards[i] + \
+                    self.gamma * self.returns[i + 1] * self.masks[i]
+
+    def update(self):
+        with chainer.no_backprop_mode():
+            _, next_value = self.model.pi_and_v(self.states[-1])
+            next_value = next_value.data
+
+        self._compute_returns(next_value)
+        pout, values = \
+            self.model.pi_and_v(chainer.Variable(
+                self.states[:-1].reshape([-1] + list(self.obs_shape))))
+
+        actions = chainer.Variable(
+            self.actions.reshape([-1] + list(self.action_shape)))
+        if isinstance(pout, distribution.CategoricalDistribution):
+            dist_entropy = F.mean(pout.entropy)
+        elif isinstance(pout, distribution.GaussianDistribution):
+            log_prob = pout.log_prob(actions, keepdims=True)
+            dist_entropy = F.mean(F.sum(log_prob, axis=-1))
```

Comment:
Why not using `GaussianDistribution.entropy`?

Reply:
I'd been misunderstanding meaning of differential entropy.
Modified to use pout.entropy(GaussianDistribution.entropy) in case of continuous action space.